src/jira.js

```diff
@@ -735,18 +735,39 @@ export default class JiraApi {
     }));
   }
 
+  /** Update Jira component
+   * [Jira Doc](http://docs.atlassian.com/jira/REST/latest/#api/2/component-updateComponent)
+   * @name updateComponent
+   * @function
+   * @param {string} componentId - the Id of the component to update
+   * @param {object} component - Properly Formatted Component
+   */
+  updateComponent(componentId, component) {
+    return this.doRequest(this.makeRequestHeader(this.makeUri({
+      pathname: `/component/${componentId}`,
+    }), {
+      method: 'PUT',
+      followAllRedirects: true,
+      body: component,
+    }));
+  }
+
   /** Delete component from Jira
-   * [Jira Doc](http://docs.atlassian.com/jira/REST/latest/#id290791)
+   * [Jira Doc](https://developer.atlassian.com/cloud/jira/platform/rest/v2/#api-api-2-component-id-delete)
    * @name deleteComponent
    * @function
-   * @param {string} componentId - the Id of the component to delete
+   * @param {string} id - The ID of the component.
+   * @param {string} moveIssuesTo - The ID of the component to replace the deleted component.
+   *                                If this value is null no replacement is made.
    */
-  deleteComponent(componentId) {
+  deleteComponent(id, moveIssuesTo = null) {
+    const query = moveIssuesTo ? { moveIssuesTo } : null;
     return this.doRequest(this.makeRequestHeader(this.makeUri({
-      pathname: `/component/${componentId}`,
+      pathname: `/component/${id}`,
```

Comment:
I would not rename the variable here especially if we have `updateComponent(componentId, component)` above.

Reply:
You are right. Unfortunately, the Jira API has been rewritten more than once since this project started. I was thinking to propose a complete rewrite of `jira.js` to bring it in line with the current documentation. Any thoughts on it?

Reply:
I think we can focus on adding more methods and patching current issues for now. I will refrain myself have from rewriting anything, especially as new maintainer 😅 The surface of the library is growing and I am still exploring the source. If you are super confident about a rewrite, go ahead!